Dockerfile

```diff
@@ -1,13 +1,36 @@
 # For building gcr.io/forgerock-io/secret-agent:latest
 FROM gcr.io/forgerock-io/ds-empty/pit1:latest AS ds
 
-FROM golang:1.13.10-alpine3.11 AS build
-ENV CGO_ENABLED=0 GOOS=linux GOARCH=amd64
-WORKDIR /go/src/github.com/ForgeRock/secret-agent
-COPY go.mod go.sum /go/src/github.com/ForgeRock/secret-agent/
+# Build the manager binary
+FROM golang:1.13-alpine as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
-COPY . /go/src/github.com/ForgeRock/secret-agent/
-RUN go build
+
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+COPY pkg/ pkg/
```

Comment:
Is there a reason here not to just do `COPY . .`? The saved steps saves build time. Also, there's `.dockerignore` if there's things you want to keep out.